### PR TITLE
Feature/action on destroy

### PR DESCRIPTION
**Pull request by qxmips**
Implements https://github.com/pablovarela/terraform-provider-slack/issues/79

### CODE REVIEW

File: docs/resources/conversation.md [hunk 1]
```diff
@@ -43,6 +43,18 @@ resource "slack_conversation" "test" {
...
 
+```hcl
+resource "slack_conversation" "nonadmin" {
+  name              = "my-channel01"
+  topic             = "The channel won't be archived on destroy"
+  permanent_members = []
+  is_private        = true
+  action_on_destroy = "none"
+  
+
```
Comment (pablovarela) [inline docs/resources/conversation.md:54]: Remove blank lines

File: slack/resource_conversation.go [hunk 2]
```diff
@@ -246,12 +264,19 @@ func resourceSlackConversationDelete(ctx context.Context, d *schema.ResourceData
 	client := m.(*slack.Client)
 
 	id := d.Id()
-	err := archiveConversationWithContext(ctx, client, id)
-	if err != nil {
-		if err.Error() == "channel_not_found" {
-			return diags
+	action := d.Get("action_on_destroy").(string)
+	switch action {
```
Comment (pablovarela) [inline slack/resource_conversation.go:268]: even though you are using a validate function in the schema I would prefer if a `default` clause was included here as well, just to be more defensive.

```go
switch action {
    default: 
         return diag.Errorf("unknown action_on_destroy value. Valid values are %v", conversationActionValidValues)
}
```

This helps for example if you added a new valid value that the schema validator accepts but it is not added to the switch clause, or misspelled. 

You would have `conversationActionValidValues` as a var:

```go
var (
	conversationActionValidValues = []string{
		conversationActionOnDestroyNone,
		conversationActionOnDestroyArchive,
	}
	validateConversationActionOnDestroyValue = validation.StringInSlice(conversationActionValidValues, false)
)
```

File: slack/resource_conversation.go [hunk 3]
```diff
@@ -246,12 +264,19 @@ func resourceSlackConversationDelete(ctx context.Context, d *schema.ResourceData
...
+
+	case conversationActionOnDestroyNone:
+		fmt.Sprintf("Do nothing on Conversation: %s (%s)", id, d.Get("name"))
```
Comment (pablovarela) [inline slack/resource_conversation.go:271]: This can be replaced with 
```go
	case conversationActionOnDestroyNone:
		diags = append(diags, diag.Diagnostic{
			Severity: diag.Warning,
			Summary:  fmt.Sprintf("conversation %s (%s) won't be archived on destroy", id, d.Get("name")),
			Detail:   fmt.Sprintf("action_on_destroy is set to %s which does not archive the conversation ", conversationActionOnDestroyNone),
		})
```

or something similar

**Review by pablovarela — CHANGES_REQUESTED** [review]
Thank you very much for your contribution! Much appreciated.

You did not add tests for the new behaviour `action_on_destroy = none`. I am not too worried about that because the default behaviour `action_on_destroy = archive` is tested by the current tests, and those are passing. I will add tests for the `none` case once we merge this.

There are some linting issues raised by the linter and some minor modifications I requested. Once addressed I will merge the PR.

File: docs/resources/conversation.md (hunk 1 above)
Comment (qxmips) [reply docs/resources/conversation.md]: Hi Pablo, thank you for accepting the changes.  and Thank you for helping with the code. I just haven't figured out a way to test this option right. I have added the changes and tested them. Please review.

File: slack/resource_conversation.go (hunk 2 above)
Comment (qxmips) [reply slack/resource_conversation.go:271]: fixed

File: slack/resource_conversation.go (hunk 3 above)
Comment (qxmips) [reply slack/resource_conversation.go]: fixed

**Review by pablovarela — APPROVED** [review]
lgtm

**Comment by qxmips** [comment]
@pablovarela Hi, when do you plan to release the new provider version?
